### DISCUSSION TITLE
Users created using Dashboard "Add User' button can sign in, but users created using JS API cannot sign in.

### DISCUSSION THREAD

**Discussion by varshithmee** [supabase/supabase#19158] [Questions]
When I create a user using signUp callback, the response returned contains a session and a user object, and the user is added to the auth table in Supabase. But, when I try to log in using the same credentials,  it returns a 400 Error ("no API key was found").   

However,  when I add a user using
![image](https://github.com/supabase/supabase/assets/64714741/852b76e4-0bff-4391-8ddf-f6bbec6c51b3)
the same user can log in! 

Things I have tried:

- Rechecked Site URL
- ReAdded  URL and ANON_KEY
- switched  from  @supabase/ssr to @supabase/supabase-js and vice versa

SignUp function
```
async function signUpWithEmail() {

    if (signUpPassword !== signUpPasswordConfirm) {
      setSignUpError("Passwords do not match!")
      return
    }

    if (signUpPassword.length < 8) {
      setSignUpError("Password must be at least 8 characters long!")
      return
    }

    if (date === undefined) {
      setSignUpError("Please enter your birthday!")
      return
    }

    //check if age is atleast 16 
    const today = new Date()
    const birthDate = new Date(date)
    let age = today.getFullYear() - birthDate.getFullYear()
    const m = today.getMonth() - birthDate.getMonth()
    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
      age--
    }
    if (age < 16) {
      setSignUpError("You must be atleast 16 years old to use Wessenger!")
      return
    }



    const { data, error } = await supabase.auth.signUp(
      {
        email: signUpEmail,
        password: 'example-password',
        options: {
          data: {
            name: signUpName,
            dateOfBirth: date,
          }
        }
      }
    )

    if (error) {
      setSignUpError(error.message)
      return
    }
    console.log(data)
    if (data) {
      //check
      
      //send data to user table
      const firstName = signUpName.split(" ")[0]
      const lastName = signUpName.split(" ").slice(1).join(" ")
      console.log(lastName)

      const { data, error } = await supabase.from('users').insert([
        { email: signUpEmail, username: signUpUsername, first_name: firstName, last_name: lastName, date_of_birth: date }
      ])

      if (error) {
        setSignUpError(error.message)
        return
      }
    }
    setSignUpError("")
    setSignUpSuccess(true)
  }
```

Sign In Function 
```
async function signInWithEmail() {

    const { data, error } = await supabase.auth.signInWithPassword({
      email: signInEmail,
      password: signInPassword
    })

    if (error) {
      setSignInError(true)
    }

    if (data) {
      setSignInError(false)
      setSignInSuccess(true)
      console.log(data)
    }
  }
```

**Comment by GaryAustin1** [comment]
Do you have confirm email set in the dashboard email provider settings?

If so there will be no user session returned from signUp.  You will get some user info but not a session.  You are then redirected to SiteURL in your case as you don't have an emailRedirectTo parameter in the signUp call,  when the user clicks the email link.

Your error though seems to be talking about your createClient not having an APIkey possibly...

**Reply by varshithmee** [reply]
No, I disabled confirm email in provider settings. 

I get user info and a session object in data from signUp.

I followed the createClient guide for JS. My client has a URL and ANON key. I can add, read, and delete from the database.  It is only the signInWithPassword callback that is returning was a server error

**Reply by GaryAustin1** [reply]
Seems very odd with your description of different users and the error apikey error occurring based on their signup method.
The network request contains the correct apikey header (anon not user)?  If not on browser you might try a select call right before.  But none of this makes sense if one user can succeed and the other can't.
Are you using next.js 13/14?  Although not seen its caching impact auth stuff.
Might be the error is not clear it what it means, as apikey should be the first call after createClient not have an apikey header correct.
I've got nothing else for you tonite.

**Reply by varshithmee** [reply]
I am using Tauri with NextJS. 

The https request for signIn contains apikey header and authorization header with "Bearer {apikey}".

But why would the client work for editing the database if I had the wrong apikey. I can also signup without any problems.

Thank you so much for you time though :-)
